.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,181 @@
+name: Benchmark Xdebug performance
+
+on:
+    schedule:
+        # Every Sunday at 03:00 UTC
+        - cron: '0 3 * * 0'
+    workflow_dispatch: 
+    pull_request:
+        types: [opened, edited, reopened, synchronize]
+
+jobs:
+    run-benchmark:
+        if: |
+            github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' ||
+            (
+                github.event_name == 'pull_request' &&
+                (
+                    startsWith(github.event.pull_request.title, 'perf:') ||
+                    startsWith(github.event.pull_request.title, 'Perf:') ||
+                    startsWith(github.event.pull_request.title, 'Performance:') ||
+                    startsWith(github.event.pull_request.title, 'performance:')
+                )
+            )
+        runs-on: ubuntu-latest
+        strategy:
+            fail-fast: false
+            matrix:
+                command: ["bench", "symfony", "rector"]
+                php: ["8.0", "8.1", "8.2", "8.3", "8.4", "8.5"]
+                xdebug_mode: ["no", "off", "develop", "coverage", "debug", "gcstats", "profile", "trace"]
+        steps:
+            -   uses: actions/checkout@v4
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: "${{ matrix.php }}"
+                    coverage: none
+                    tools: composer:v2.2.21
```

Comment:
We need to use a specific version of Composer because we are using a specific version of the Symfony Demo which is not compatible with the latest versions of Composer